### PR TITLE
Enable multiple actions in the case list

### PR DESCRIPTION
**Pull request by phillipm**
Add the ability have more than one action to a case list. Needed for having a 'case migration' action along with other actions like 'register patient'

cross-requested with https://github.com/dimagi/commcare/pull/248

contains commits from https://github.com/dimagi/commcare-odk/pull/1024

screen

### CODE REVIEW

File: app/src/org/commcare/android/adapters/EntityListAdapter.java [hunk 1]
```diff
@@ -203,13 +212,14 @@ public int getItemViewType(int position) {
      */
     @Override
     public View getView(int position, View convertView, ViewGroup parent) {
-        if (actionEnabled && position == actionPosition) {
+        if (actionsCount > 0 && position >= actionsStartPosition) {
```
Comment (amstone326) [inline app/src/org/commcare/android/adapters/EntityListAdapter.java]: Might be nice to factor these 2 cases out into their own methods, e.g. `getActionView()` and `getNormalView()`

Comment (phillipm) [inline app/src/org/commcare/android/adapters/EntityListAdapter.java]: :+1:

File: app/src/org/commcare/dalvik/activities/EntitySelectActivity.java [hunk 2]
```diff
@@ -619,7 +621,7 @@ protected void onDestroy() {
     @Override
     public void onItemClick(AdapterView<?> listView, View view, int position, long id) {
         if (id == EntityListAdapter.SPECIAL_ACTION) {
-            triggerDetailAction();
+            triggerDetailAction(position - (adapter.getCount() - 1));
```
Comment (amstone326) [inline app/src/org/commcare/dalvik/activities/EntitySelectActivity.java]: It's a little bit obfuscated what these counts represent / what you're trying to calculate here; maybe worth adding a comment.

**Comment by amstone326** [comment]
@phillipm Suggested a comment, good to merge after.

File: app/src/org/commcare/dalvik/activities/EntitySelectActivity.java (hunk 2 above)
Comment (phillipm) [inline app/src/org/commcare/dalvik/activities/EntitySelectActivity.java]: :ok_hand:

**Pull request merged**